instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/messaging/MessagingAttributesGetter.java

```diff
@@ -25,20 +25,43 @@ public interface MessagingAttributesGetter<REQUEST, RESPONSE> {
   @Nullable
   String getDestination(REQUEST request);
 
+  @Nullable
+  String getDestinationTemplate(REQUEST request);
+
   boolean isTemporaryDestination(REQUEST request);
 
+  boolean isAnonymousDestination(REQUEST request);
+
   @Nullable
   String getConversationId(REQUEST request);
 
   @Nullable
-  Long getMessagePayloadSize(REQUEST request);
+  @Deprecated
+  default Long getMessagePayloadSize(REQUEST request) {
+    return null;
+  }
+
+  @Nullable
+  @Deprecated
+  default Long getMessagePayloadCompressedSize(REQUEST request) {
+    return null;
+  }
 
   @Nullable
-  Long getMessagePayloadCompressedSize(REQUEST request);
+  Long getMessageBodySize(REQUEST request);
+
+  @Nullable
+  Long getMessageEnvelopeSize(REQUEST request);
 
   @Nullable
   String getMessageId(REQUEST request, @Nullable RESPONSE response);
 
+  @Nullable
+  String getClientId(REQUEST request);
```

Comment:
this is currently used only by kafka and was previously set in the kafka extractors